docs/netlify-dev.md

```diff
@@ -157,7 +154,14 @@ As for which port to use while doing local development in Netlify Dev, always lo
    └──────────────────────────────────────────────────────────────┘
 ```
 
-</details>
+**Specifying custom ports for Netlify Dev**
+
+Netlify Dev allows you to specify the following parameters for port as both flags and in config file (`netlify.toml` etc.):
+* `port`: The port for the main Netlify Dev server, the one you'll open in browser.
+* `targetPort`: The port for your application server or site generator.
+* `functionsPort`: The port for Netlify Functions server. This server is accessed internally within Netlify Dev, you shouldn't need to access it directly. And thus shouldn't need to change this port. You can access your functions at Netlify Dev main server port like so: `http://localhost:<port>/.netlify/functions/<your-function-slug>`.
+
+Netlify Dev tries to acquire these ports but if any of them is not available (already in use by another application), it assigns a random port instead of that.
```

Comment:
Shouldn't we fail if the user explicitly sets a port but it is not available? 
I'm assuming users set a specific port since they want to use it later on in their code/login, so assigning a random one might create issues.

Don't think it is a blocker for the PR, just something to consider

Reply:
Yes please. Fail fast when the port is taken.

Else most people will start their server, try to use the port they specify, get confused on why it is not working, try clearing their cache or something, and then finally check what port it actually started on.

Reply:
Can be discussed under #767.